spec/std/spec/expectations_spec.cr

```diff
@@ -169,8 +169,204 @@ describe "expectations" do
   end
 
   describe "expect_raises" do
-    it "pass if raises MyError" do
+    it "passes if expected message equals actual message and expected class equals actual class" do
       expect_raises(Exception, "Ops") { raise Exception.new("Ops") }
     end
+
+    it "passes if expected message equals actual message and expected class is an ancestor of actual class" do
+      expect_raises(Exception, "Ops") { raise ArgumentError.new("Ops") }
+    end
+
+    it "passes if expected message is a substring of actual message and expected class equals actual class" do
+      expect_raises(Exception, "Ops") { raise Exception.new("Black Ops") }
+    end
+
+    it "passes if expected message is a substring of actual message and expected class is an ancestor of actual class" do
+      expect_raises(Exception, "Ops") { raise ArgumentError.new("Black Ops") }
+    end
+
+    it "passes if expected regex matches actual message and expected class equals actual class" do
+      expect_raises(Exception, /Ops/) { raise Exception.new("Black Ops") }
+    end
+
+    it "passes if expected regex matches actual message and expected class is an ancestor of actual class" do
+      expect_raises(Exception, /Ops/) { raise ArgumentError.new("Black Ops") }
+    end
+
+    it "passes if given no message expectation and expected class equals actual class" do
+      expect_raises(Exception) { raise Exception.new("Ops") }
+    end
+
+    it "passes if given no message expectation and expected class is an ancestor of actual class" do
+      expect_raises(Exception) { raise ArgumentError.new("Ops") }
+    end
+
+    it "passes if given no message expectation, actual message is nil and expected class equals actual class" do
+      expect_raises(Exception) { raise Exception.new(nil) }
+    end
+
+    it "passes if given no message expectation, actual message is nil and expected class is an ancestor of actual class" do
+      expect_raises(Exception) { raise ArgumentError.new(nil) }
+    end
+
+    it "fails if expected message does not equal actual message and expected class equals actual class" do
+      expect_raises(Exception, "Ops") { raise Exception.new("Hm") }
+    rescue Spec::AssertionFailed
+      # success
+    else
+      raise "expected Spec::AssertionFailed but nothing was raised"
```

Comment:
That's redundant due to the `expect_raises` usage, which handles that already.

ditto elsewhere

Reply:
Hm, we test here specifically `expect_raises` and if it's broken and it doesn't fail test (that's no `Spec::AssertionFailed` exception raised) the test is supposed to fail. But without this `else` branch the test will succeed.